### PR TITLE
Fix code style issues and code smells in logstash

### PR DESCRIPTION
**Pull request by sjinks**
## Description

This PR fixes code style compliance issues found by `phpcs` and some of the code smells found by `sonarlint`.

### Fix code style compliance issues in `logstash` directory

  * Add phpcs:ignore where appropriate to suppress false positives or known good code;
  * Fix code smells detected by `sonarlint`.

## Checklist

Please make sure the items below have been covered before requesting a review:

- [x] This change works and has been tested locally (or has an appropriate fallback).
- [ ] This change works and has been tested on a Go sandbox.
- [ ] This change has relevant unit tests (if applicable).
- [ ] This change has relevant documentation additions / updates (if applicable).
- [x] I've created a changelog description that aligns with the provided examples.

## Steps to Test

Make sure that the CI passes.

### CODE REVIEW

File: logstash/class-logger.php [hunk 1]
```diff
@@ -306,13 +306,14 @@ protected static function parse_params( array $params ) : array {
 		];
 
 		if ( ! isset( $params['file'] ) && ! isset( $params['line'] ) ) {
+			// phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_debug_backtrace
 			$backtrace = debug_backtrace( DEBUG_BACKTRACE_IGNORE_ARGS, 1 );
 
-			if ( isset( $backtrace[0] ) && isset( $backtrace[0]['file'] ) ) {
+			if ( isset( $backtrace[0]['file'] ) ) {
```
Comment (sjinks) [inline logstash/class-logger.php:312]: Here and below: the second `isset()` checks both `$backtrace[0]` and `$backtrace[0]['file']`; the first `isset()` check is superfluous.

File: logstash/class-logger.php [hunk 2]
```diff
@@ -306,13 +306,14 @@ protected static function parse_params( array $params ) : array {
...
 				$default_params['file'] = $backtrace[0]['file'];
 			}
 
-			if ( isset( $backtrace[0] ) && isset( $backtrace[0]['line'] ) ) {
+			if ( isset( $backtrace[0]['line'] ) ) {
```
Comment (sjinks) [inline logstash/class-logger.php:316]: **Possible bug:** this method is protected; `$backtrace[0]` refers to the caller. The caller will be `Logger::log2logstash()`. Shouldn't we check the first frame, not the zeroth one?

Comment (rebeccahum) [reply logstash/class-logger.php:316]: Pinging @hanifn for your input on this based on your PR at https://github.com/Automattic/vip-go-mu-plugins/pull/1530/files.

**Review by rebeccahum — APPROVED** [review]
LGTM!

**Pull request merged**